torch/csrc/api/include/torch/nn/modules/embedding.h

```diff
@@ -91,15 +43,16 @@ class Embedding : public torch::nn::ModuleHolder<EmbeddingImpl> {
  public:
   using torch::nn::ModuleHolder<EmbeddingImpl>::ModuleHolder;
 
-  static Embedding from_pretrained(const torch::Tensor& embeddings, c10::optional<EmbeddingOptions> options = c10::nullopt, bool freeze = true) {
+  static Embedding from_pretrained(const torch::Tensor& embeddings, EmbeddingOptions options = {}, bool freeze = true) {
     TORCH_CHECK(embeddings.dim() == 2, "Embeddings parameter is expected to be 2-dimensional");
-    if (options != c10::nullopt) {
-      TORCH_CHECK((*options).num_embeddings() == embeddings.size(0), "Expects options.num_embeddings to be ", embeddings.size(0) , "but found ", (*options).num_embeddings());
-      TORCH_CHECK((*options).embedding_dim() == embeddings.size(1), "Expects options.embeddings_dim to be ", embeddings.size(1) , "but found ", (*options).embedding_dim());
-    } else {
-      options = EmbeddingOptions(embeddings.size(0), embeddings.size(1));
+    if (options.num_embeddings()) {
+      TORCH_WARN(*options.num_embeddings() == embeddings.size(0), "`num_embeddings` options parameter is ignored in `torch::nn::Embedding::from_pretrained`.");
     }
-    Embedding embedding((*options)._weight(embeddings));
+    if (options.embedding_dim()) {
+      TORCH_WARN(*options.embedding_dim() == embeddings.size(1), "`embedding_dim` options parameter is ignored in `torch::nn::Embedding::from_pretrained`.");
```

Comment:
I believe we don't need to check `*options.embedding_dim() == embeddings.size(1)`, because `options.embedding_dim()` would just be ignored. We can write:
```suggestion
      TORCH_WARN("`embedding_dim` options parameter is ignored in `torch::nn::Embedding::from_pretrained`.");
```